### ISSUE TITLE
[UX] Revamp the confirmation pages for updating module/themes/layouts

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3957]
**Description of the need**

1. Head to `/admin/modules/update`
2. Select an update, and click on the "Download these updates" button...

Here's what the confirmation page currently looks like:

<img width="572" alt="Screen Shot 2019-08-03 at 6 14 49 am" src="https://user-images.githubusercontent.com/2423362/62406545-62562980-b5b6-11e9-98cd-70cbccb04787.png">

**Proposed solution**

- Make "Back up your database and site before you continue" a proper warning
- Add a "Cancel" button

<img width="620" alt="Screen Shot 2019-08-07 at 8 38 09 am" src="https://user-images.githubusercontent.com/2423362/62597444-c4809880-b8ee-11e9-8eab-52a1ba2468be.png">

---
PR by @klonos: https://github.com/backdrop/backdrop/pull/2804

**Comment by klonos** [comment]
...while at it, I have updated the link to point to https://backdropcms.org/upgrade instead of the current http://drupal.org/node/22281 😉

I am wondering if we should have some sort of constant for the message, so that it can be reused (in places like #512 for example). This would also make it so that translators translate things in a single place, and we update this message in a single place in the future if need be.

**Comment by stpaultim** [comment]
Can you add a quick comment on how to test this? I assume I can't do it in Sandbox.

**Comment by klonos** [comment]
> I assume I can't do it in Sandbox.

Yes you can 🙂 ...just head to Functionality -> Update modules (`/admin/modules/update`) ...there is currently a known (useful in this case) bug which lists the current stable release (1.13.2 at the time of writing this) as an update of the current dev release (1.14.x-dev). Simply tick the checkbox next to Backdrop core, and click the "Download these updates" button ...once the download is complete, you should be taken to this page 😉

**Comment by klonos** [comment]
...note: we could use a "Cancel" link in this form, but we need to account for destination, so that the user is taken back to the page where they started (modules/themes/layouts updates) ...is that out of scope for this issue, or can we sneak that in while at it?

**Comment by klonos** [comment]
Adding the proper link/redirect to the "Cancel" button turned out to be trickier than expected...

Setting `$_GET['destination']` on the submit handler for `installer_manager_update_form_submit()` dit not work, as that was blanked by the time you reach `installer_manager_update_ready_form()` (I am guessing that the batch has something to do with that).

I have also tried appending the destination to the form action via a `$form['#action'] .= '?destination=' . drupal_get_destination();`, but that did not work either.

I then also tried setting `'url_options' => array('query' => array('destination' => 'admin/modules')),` in the batch definition array, but that does not seem to work either 😞

I ended up storing the project type in `$_SESSION` instead, which works.

**Comment by klonos** [comment]
...this has been discussed around the web in various articles. Here's one example: https://support.acquia.com/hc/en-us/articles/360004334093-Removing-large-files-from-Git-without-losing-history
